suite-native/module-settings/src/components/CurrencySelector.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { Select, SelectValue } from '@suite-native/atoms';
+import { CurrencyType, fiatCurrenciesMap } from '@suite-common/suite-config';
+
+import { selectCurrency, setCurrency } from '../slice';
+import { transformFiatCurrencyToSelectItem } from '../utils';
+
+export const CurrencySelector = () => {
+    const selectedCurrency = useSelector(selectCurrency);
+    const dispatch = useDispatch();
+
+    const handleSelectCurrency = (value: SelectValue) => {
+        dispatch(setCurrency(value as CurrencyType));
+    };
+
+    const fiatCurrencies = Object.values(fiatCurrenciesMap).map(transformFiatCurrencyToSelectItem);
```

Comment:
This should be moved before component, it's static config and it doesn't depend on anything here.

`fiatCurrencies` => `fiatCurrenciesItems` because it's specific list of items for select.

Also one more thing, we are going to use pipes everywhere instead of chains:
```tsx
import { A, D, pipe } from '@mobily/ts-belt';

const fiatCurrenciesItems = pipe(
    fiatCurrencies,
    D.values,
    A.map(transformFiatCurrencyToSelectItem),
)
```
I know these kind of syntax could be unfamiliar to you and it may feel strange but pipes are very well known thing from FP world and readability is much better when get use to it and especially when need to chain a lot of things. Another benefit is that in ts-belt toolkit has lot of useful functions and you can use them to simplify things a lot and it's hard to combine them with chains.